src/main/java/com/powsybl/openloadflow/network/LfContingency.java

```diff
@@ -79,7 +87,7 @@ public double getActivePowerLoss() {
         }
 
         // check if contingency split this network into multiple components
-        if (branches.isEmpty()) {
+        if (branches.isEmpty() && propagatedContingency.getShuntIdsToLose().isEmpty()) {
```

Comment:
@geofjamg here maybe my check is not so clean... it was designed for branch contingencies only. Could you maybe improve it ?

Reply:
Looks good, but probably an issue with other injection contingency